test/HarpoonKrakenTests.cpp

```diff
@@ -30,6 +30,51 @@ void* HarpoonKrakenTests::SendHello(void* arg) {
    return nullptr;
 }
 
+
+namespace {
+   struct HelloAbort {
+      std::string address;
+      std::atomic<size_t>  abortAt;
+      explicit HelloAbort(std::string addressQ)
+         : address(addressQ)
+         , abortAt(0)
+      {}
+   };
+} // anonymous
+void* HarpoonKrakenTests::SendHelloExpectCancel(void* arg) {
+   using AbortNotifyPtr = std::shared_ptr<HelloAbort>;
+   std::shared_ptr<HelloAbort> helloAbort = *(reinterpret_cast<AbortNotifyPtr*>(arg));
+
+
+   Kraken server;
+   server.SetLocation(helloAbort->address);
+   server.MaxWaitInMs(1000); // 1 second
+
+   uint8_t h = 'h';
+   uint8_t e = 'e';
+   uint8_t l = 'l';
+   uint8_t o = 'o';
+   uint8_t end = '\0';
+   std::vector<uint8_t> hello{h, e, l, l, o, end};
+
+   signed int counter = -1; // 1 off 
+   auto status = Kraken::Battling::CONTINUE;
+   for (auto c : hello) {
+      status = server.SendTidalWave({c});
+      if (status == Kraken::Battling::CANCEL) {
+         helloAbort->abortAt.store(counter);
+         LOG(INFO) << "Received cancel at count: " << counter;
```

Comment:
Not sure I understand @john-gress. That's exactly what I am doing (setting a boolean) , line: 65

Please suggest with pseudo code if you see how to improve it